internals/overlord/tlsstate/package_test.go

```diff
@@ -0,0 +1,138 @@
+// Copyright (C) 2025 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package tlsstate_test
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"testing"
+	"time"
+
+	. "gopkg.in/check.v1"
+)
+
+// Hook up check.v1 into the "go test" runner.
+func Test(t *testing.T) { TestingT(t) }
+
+type tlsSuite struct{}
+
+var _ = Suite(&tlsSuite{})
+
+// GetFakeTime converts a simple string date to the time.Time type for testing purposes.
+func (ts *tlsSuite) GetFakeTime(c *C, date string) time.Time {
+	layout := "2006-01-02"
+	now, err := time.Parse(layout, date)
+	c.Assert(err, IsNil)
+	return now
+}
+
+// testTLSVerifiedClient performs a client TLS connection without server certificate
+// signature verification.
+func (ts *tlsSuite) testTLSInsecureClient(c *C, clock time.Time) ([]*x509.Certificate, error) {
+	return ts.testTLSClient(c, nil, clock)
+}
+
+// testTLSVerifiedClient performs a client TLS connection with server certificate
+// signature verification.
+func (ts *tlsSuite) testTLSVerifiedClient(c *C, ca *x509.Certificate, clock time.Time) ([]*x509.Certificate, error) {
+	return ts.testTLSClient(c, ca, clock)
+}
+
+func (ts *tlsSuite) testTLSClient(c *C, ca *x509.Certificate, clock time.Time) ([]*x509.Certificate, error) {
+	var serverCerts []*x509.Certificate
+
+	certPool := x509.NewCertPool()
+	if ca != nil {
+		certPool.AddCert(ca)
+	}
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				MinVersion:         tls.VersionTLS13,
+				MaxVersion:         tls.VersionTLS13,
+				InsecureSkipVerify: true,
+				VerifyConnection: func(ts tls.ConnectionState) error {
```

Comment:
This is unfortunate shadowing of `ts *tlsSuite` above. Can we call this something else, `connState` maybe?

Reply:
Since it is only used once on the very next line, I just made it ```cs```. Resolving.